### ISSUE TITLE
Property value referencing function in this does not work

### ISSUE THREAD

**Issue by JeremybellEU** [Polymer/polymer#3004]
The value field of a property can be a function. When this is a direct function or reference to a function defined in an outer-scope, it works. However, when a function in `this` (referencing the Polymer object) is passed it does not work.

http://jsbin.com/zipodeduzi/edit?html,js,console,output to reproduce.

Expected console output:
"called"
"called another"

Actual console output:
"called"

Expected page contents:
"asdfanother"

Actual page contents:
"asdf"

Reported by @tbone587

**Comment by JeremybellEU** [comment]
Still thinking on why this doesn't work. I think the `this` does not reference the Polymer object, but the property object. However, when trying that out (adding the function after the `value` field), it does not work either.

**Comment by arthurevans** [comment]
I think what you're seeing here is that you're passing an object with a `this` reference to the `Polymer()` function. It's _not_ inside a function closure. You're calling `Polymer()` from the global scope so `this == window`. `window.functionWithAnotherName` is undefined, so that's the default value for `another`.

Here's another jsbin, where you can see that calling `this.functionName` matches a function in the global scope, and that setting a property to `this` gives you a reference to `window`.

http://jsbin.com/rumaqu/1/edit?html,js,console,output

To put it another way: if you want per-instance initialization for a property, `value` needs to be a `Function` object _when you call `Polymer(`)_, 

You can, however, do:

    value: function() { return this.someOtherMethod(); }

**Comment by azakus** [comment]
Yep, @arthurevans is correct. Your use of `this` here will be `window` when `this.functionWithAnotherName` is dereferenced, as the `this` of executing scripts is `window.

Here's a fixed jsbin for you: http://jsbin.com/pobusa/edit?html,console,output

**Comment by JeremybellEU** [comment]
Thanks for the extensive explanation! I PM'd Tbone.
